### PR TITLE
Add a News Carousel

### LINKED ISSUE #62: Implement the carousel component
When adds the carousel, removes the about section

### PR DESCRIPTION
**Pull request by abraaovalentim**
This PR introduces the creation of the NewsCarousel component and its integration into the new CarouselSection. The NewsCarousel dynamically displays news items with an auto-advancing feature and interactive navigation buttons. 

How to Test:
Add News in CMS
Check if the section appears on the main page with only one News
Add more News
Veirfy the news transition

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -15,6 +16,7 @@ export default function Home() {
         <OcaImage id="logo-oca" />
       </LogoSection>
       <AboutSection />
```
Comment (RodrigoEC) [inline src/app/page.tsx]:
```suggestion
```

File: src/components/CarouselSection/CarouselSection.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const { loadData } = useContext(CMSContext);
+  const [newsItems, setNewsItems] = useState<INews[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  const loadNews = async () => {
+    const fetchedNewsItems = await loadData("news");
+    if (fetchedNewsItems.length > 0) {
+      setNewsItems(fetchedNewsItems as unknown as INews[]);
+    } else {
+      setNewsItems([
```
Comment (RodrigoEC) [inline src/components/CarouselSection/CarouselSection.tsx]: I think it's best if we added this "default" value as the initial value of the newItems state, with it we won't need the "else" statement

File: src/components/CarouselSection/NewsCarousel/NewsCarousel.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  z-index: 1;
+  align-self: center;
+  margin-top: auto;
+`;
+
+export const NewsTitle = styled.h2`
+  background: none;
```
Comment (RodrigoEC) [inline src/components/CarouselSection/NewsCarousel/NewsCarousel.styles.tsx]:
```suggestion
```

File: src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  loading,
+}: {
+  newsItems: INews[];
+  loading: boolean;
+}) => {
+  const [index, setIndex] = useState(0);
+  const length = 3;
+  let handler: NodeJS.Timeout;
+
+  const handleNext = (event?: { stopPropagation: () => void } | undefined) => {
```
Comment (RodrigoEC) [inline src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx]: We could refactor this function to be like a handle update, that receives either the operation string, like "next", or the function itself

File: src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    event?.stopPropagation();
+    setIndex((index + 1) % length);
+  };
+  const handlePrev = (event?: { stopPropagation: () => void }) => {
+    event?.stopPropagation();
+    setIndex((index - 1 + length) % length);
+  };
+
+  const funcDebounce = () => {
```
Comment (RodrigoEC) [inline src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx]:
```suggestion
  const updateDebound = () => {
```

File: src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    handler = setTimeout(() => {
+      handleNext();
+    }, 7000);
+  };
+
+  const redirectUrl = () => {
```
Comment (RodrigoEC) [inline src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx]: Instead of using a function to redirect, it's best if we use a link (<a>) tag

**Review by RodrigoEC — CHANGES_REQUESTED** [review]
Great Work! Creating a carousel is rather boring, but you shined. Just added some refactoring suggestions.

image

Rebeca would be proud

**Review by RodrigoEC — APPROVED** [review]

**Comment by cilasmarques** [comment]
I found a bug in smartphone screens.
image

**Comment by abraaovalentim** [comment]
> I found a bug in smartphone screens. image

well here I don't have this error
Captura de tela de 2024-08-06 15-21-33

**Comment by cilasmarques** [comment]
> > I found a bug in smartphone screens. image
> 
> well here I don't have this error Captura de tela de 2024-08-06 15-21-33

I checked here and it's because of the browser

File: src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        <LoadingIcon id="loading" />
+      ) : (
+        <>
+          <NewsImage
+            width={1600}
+            height={800}
+            src={
+              newsItems[index]?.fields.thumb.fields.file.url
+                ? `https://${newsItems[index]?.fields.thumb.fields.file.url}`
+                : ""
```
Comment (cilasmarques) [inline src/components/CarouselSection/NewsCarousel/NewsCarousel.tsx]:
```suggestion
            src={newsItems[index]?.fields.thumb.fields.file.url && `https://${newsItems[index]?.fields.thumb.fields.file.url}`}
```

**Review by cilasmarques — APPROVED** [review]

**Pull request merged**